test-case/manual/wov-script/wov-test.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+rm -rf blob
+preamble_time=0
+history_depth=0
+format=s16_le
+buffer_size=68000
+max_loop=100
+preamble_time_init=2100
+history_depth_init=1800
+str=$(amixer contents | grep "Detector")
+control=${str#*=}
+control_id=${control%%,*}
+
+
+usage(){
+        echo "Wrong parameters, please specify the test item"
+        echo "Eg: '-p' preamble_time, '-h' history_depth, '-f' format and '-s' period-size\
+        ./$0 -p 2500 -h 3000 -f s16_le -s 500"
+}
+
+# DEC to HEX
+dec2hex(){
+        printf "%x" "$1"
+}
+
+_save_def(){
+	echo "Save WoV as default configure blob:"
+	if [ ! -f detector-default-config ]; then
+		sudo sof-ctl -Dhw:0 -n "$control_id" -br -o detector-default-config
+	fi
+}
+
+_set_default(){
+	echo "Set WoV as default configure blob:"
+	sudo sof-ctl -Dhw:0 -n "$control_id" -br -s detector-default-config
+}
+
+# get the preamble time and history depth
+_get_pt_hd(){
+	_set_default
+        # save current WoV configure blob
+	sof-ctl -Dhw:0 -n "$control_id" -br -o blob
+	def_1st=x08
+	def_snd=x34
+
+}
+
+# set the preamble time & history depth
+_set_pt_hd(){
+        preamble_time_aft=$(dec2hex $preamble_time)
+	length_pt=$(echo "$preamble_time_aft" |wc -L)
+	if [ "$length_pt" -eq 3 ]; then
+		preamble_time_aft=0$preamble_time_aft
+	fi
+	pt_aft_1st=$(echo "$preamble_time_aft" |cut -c 1-2)
+	pt_aft_1st=x$pt_aft_1st
+	pt_aft_snd=$(echo "$preamble_time_aft" |cut -c 3-4)
+	pt_aft_snd=x$pt_aft_snd
+        history_depth_aft=$(dec2hex $history_depth)
+	length_hd=$(echo "$history_depth_aft" |wc -L)
+	if [ "$length_hd" -eq 3 ]; then
+		history_depth_aft=0$history_depth_aft
+	fi
```

Comment:
I don't understand this, why not just:
```shell
dec2hex () 
{ 
    printf "0x%x" "$1"
}
```
Can you share some examples of actual values?